### ISSUE TITLE
AI: Phase I - Training Pipeline

### ISSUE THREAD

**Issue by Khoile5464** [SendoAIAssistance/Client#6]
# Phase I: Training Pipeline

<img width="566" height="959" alt="Image" src="https://github.com/user-attachments/assets/563f4ebf-3198-440d-a51a-cdfede5d0e52" />

The primary goal of this phase is to train the 'Element Analysis Model', 'Planner Model' and 'Fraud Detection Model'.
### **Element Analyis Model:** [Named Entity Recognition Paper](https://arxiv.org/html/2401.10825v3)
1. **Data Preparation:**
    * **Input:** Raw natural language queries (e.g., *"Giúp chị xuất đơn số 8015..."*).
    * **Ground Truth:** Labeled JSON data containing correct entities (e.g., `OrderID`, `PaymentID`, `Name`, etc.).
2. **Encoding:** Inputs are tokenized and encoded into vector embeddings (e.g., `[0.5, 0.125, ...]`).
3. **Forward Pass:**
    * The encoded data is passed through the **Element Analysis Model**.
    * **Output:** Generates a predicted result vector/JSON.
4. **Optimiazation Loop:**
    * **Loss Calculation:** The system compares the *AI Generated Result* with the *Expected Relsult* to calculate the loss value (e.g., `Loss Result: 89.5689`).
    * **Backpropagation:** As always, the system triggers the training function to update the model's weights.
    * **Convergence:** The loop continues until the loss is minimized (Global Minima), ensuring high accuracy in entity extraction.

---
### **Planner Model Optimization (Decision Engine)** 
The Planner is responsible for breaking down user intent into a sequence of logical actions [Chain of Thoughts](https://arxiv.org/pdf/2201.11903)
1. **Data Preparation:**
    * **Trajetory Data:** Pairs of *[User Intent, Correct Sequence of Actions]*.
    * Example:
          * *Input:* "Kiểm tra đơn hàng 8544... này giúp chị."
          * *Label:* `[Extract_ID] -> [Check_Auth] -> [Call_API]`
    * **Methodology:**
          * **Supervised Fine-Tuning (SFT):** Fine-tuning a base LLM on high-quality action trajectories to enforce strict adherence to business logic.
          * **Feedback Loop:** Penalizing the model for loops (repeating actions) or hallucinating tools that do not exist.
    * **Objective:** Maximize **Action Accuracy** and minimize **Steps-to-Solution**.

### **Fraud Detection Model Training**: [Fraud Detection in Financial Transactions Paper](https://www.researchgate.net/publication/379654286_Fraud_Detection_in_Financial_Transactions)
This is a specialized binary classification model designed to analyze transation patterns returned by the API.
* **Data Preparation:** 
    * Historical transaction logs labled as `Normal` or `Fraudulent`.
    * **Feature Engineering:** Extracting features such as *Transaction Amount, Velocity (frequency), Location Mismatch. and Device Fingerprint.*
* **Model Architecture:**
    * Uses **XGBoost** or **Isolation Forest** (for anomaly detection) to calculate the Risk Score.
* **Loss Function:** Optimized to minimize False Negatives (missing a fraud case)
